realm/realm-library/src/androidTest/java/io/realm/TypeBasedNotificationsTest.java

```diff
@@ -1807,4 +1811,89 @@ public void run() {
         TestHelper.awaitOrFail(listenerWasCalledOnRealmObject);
         TestHelper.awaitOrFail(listenerWasCalledOnRealmResults);
     }
+
+    // Test modifying realmObjects in RealmObject's change listener
+    public void test_change_realm_objects_map_in_listener() throws InterruptedException {
+        final CountDownLatch finishedLatch = new CountDownLatch(2);
+        handler.post(new Runnable() {
+            @Override
+            public void run() {
+                realm = Realm.getInstance(configuration);
+
+                realm.beginTransaction();
+                // At least two objects are need to make sure list modification happen during iterating.
+                final Cat cat = realm.createObject(Cat.class);
+                final Owner owner = realm.createObject(Owner.class);
+                owner.setCat(cat);
+                realm.commitTransaction();
+
+                RealmChangeListener listener = new RealmChangeListener() {
+                    @Override
+                    public void onChange() {
+                        Cat cat = owner.getCat();
+                        boolean foundKey = false;
+                        // Keep calling addToRealmObjects just in case it won't be called in getCat in the future
```

Comment:
comment is ambiguous + the test comment state that we're modifying the RealmObject inside the listener, which we're not doing obviously?